### PR TITLE
Load individual project from firebase

### PR DESCRIPTION
**Pull request by Haider8**

### CODE REVIEW

File: src/components/Other/Project.vue [hunk 1]
```diff
@@ -1,30 +1,30 @@
 <template>
   <v-container>
-    <v-layout row wrap>
+    <v-layout row wrap v-for="proj in project" :key="proj.name">
```
Comment (iamareebjamal) [inline src/components/Other/Project.vue]: Why change to v-for?

File: src/components/Other/Project.vue [hunk 2]
```diff
@@ -5,26 +5,26 @@
         <v-card>
           <v-card-title>
             <div>
-              <h2 class="primary--text">{{ project.name }}</h2>
-              <div class="primary--text">{{ project.team }}</div>
+              <h2 class="primary--text">{{ project[0].name }}</h2>
```
Comment (iamareebjamal) [inline src/components/Other/Project.vue]: Not good. Create a getter which returns a project by ID, and if that getter returns null, then fetch from firebase and store in store

**Comment by Haider8** [comment]
Does it require more modifications?

**Comment by iamareebjamal** [comment]
Yes, see my previous comment on get by ID

File: src/components/Other/Project.vue [hunk 3]
```diff
@@ -43,25 +80,47 @@
...
   props: ['id'],
   data() {
     return {
       loading: true
     }
+  },
+  computed: {
+    loadedProject() {
+      return this.getProjectById()(this.id)
+    },
```
Comment (Haider8) [inline src/components/Other/Project.vue]: This getter getProjectById() is returning the project based on the id.

File: src/components/Other/Project.vue [hunk 4]
```diff
@@ -43,25 +80,47 @@
...
+    ...mapState('projects', ['project'])
+  },
+  methods: {
+    ...mapActions('projects', ['setProjectRef']),
+    ...mapGetters('projects', ['getProjectById'])
+  },
+  created() {
+    if (this.loadedProject == null) {
```
Comment (Haider8) [inline src/components/Other/Project.vue]: If condition such that if that getter returns null, then only fetch the project from firebase

**Comment by Haider8** [comment]
I will be thankful if you can please point out, how exactly to implement this.

**Comment by iamareebjamal** [comment]
Why are we using project[0] instead of loadedProject in some places?

**Comment by Haider8** [comment]
Because that computed property(**loadedProject**)for calling the getter and the **project** that we are calling in the mapState function have different names. I have to use the same name for both of these but it isn't working that way. I am not able to figure out how to do this.

**Comment by iamareebjamal** [comment]
OK I'll look into it

**Comment by iamareebjamal** [comment]
@Haider8 Please check the last commit

**Comment by iamareebjamal** [comment]
The dialog box is too large, make it smaller. Also, check the footer and other smaller issues and then we'll make it live

**Comment by iamareebjamal** [comment]
Do these things in other PRs

**Comment by Haider8** [comment]
@iamareebjamal checked the last commit, i couldn't have think like that. But that loadedProject getter is not very clear to me.

**Comment by iamareebjamal** [comment]
It first checks if the required project is in the projects array, then it checks if it is in the project array which was added by you for loading single project, else, it returns null

Now, the Projects.vue component checks if the getter returned null, which it will in case someone opens the single project page. If it is null, then it will dispatch the action as you wrote, and when the project array will get populated with the requested project, getProjectById function will get triggered again and this time, it will find the requested project in the project array

If someone opens a project from projects page, getter will always find the project in the projects array

**Pull request merged**